web/src/vdom.ts

```diff
@@ -2,7 +2,28 @@ import _ from "lodash";
 
 import * as blueslip from "./blueslip";
 
-export function eq_array(a, b, eq) {
+export type VDomNode = {
+    key: string;
+    eq: (other: VDomNode) => boolean;
+    render: () => string;
+    conversation: Record<string, unknown>;
```

Comment:
Also why `unknown` is being used here? I think it should be properly typed, here's the object I got by doing a simple console.log() you can take reference from this:

![Screenshot 2023-03-30 212312](https://user-images.githubusercontent.com/84276404/228894760-08d07488-8c1f-4b66-9528-a60867ff6fac.png)